static/app/views/performance/landing/widgets/widgets/singleFieldAreaWidget.tsx

```diff
@@ -39,38 +40,41 @@ export function SingleFieldAreaWidget(props: Props) {
   if (props.fields.length !== 1) {
     throw new Error(`Single field area can only accept a single field (${props.fields})`);
   }
+  const field = props.fields[0];
 
-  const Queries = useMemo(() => {
-    return {
-      chart: {
-        fields: props.fields[0],
-        component: provided => (
-          <EventsRequest
-            {...provided}
-            limit={1}
-            includePrevious
-            includeTransformedData
-            partial
-            currentSeriesName={props.fields[0]}
-            eventView={props.eventView}
-            query={props.eventView.getQueryWithAdditionalConditions()}
-            interval={getInterval(
-              {
-                start: provided.start,
-                end: provided.end,
-                period: provided.period,
-              },
-              'medium'
-            )}
-          />
-        ),
-        transform: transformEventsRequestToArea,
-      },
-    };
-  }, [props.eventView.query, props.fields[0], props.organization.slug]);
+  const chart = useMemo<QueryDefinition<DataType, WidgetDataResult>>(
```

Comment:
I'm not sure I understand why the `useMemo` here is needed. If you wrap the memorized result inside of an object like you have below, it will always trigger a re-render of the child component since the reference to `Queries` will be different even if `chart` has been memorized right?

Reply:
Re-rendering the same component is fine now, building a new component inside an object each render isn't. Since it wasn't part of the jsx in the return iirc it was being rebuilt each time.